source/fab/steps/compile_fortran.py

```diff
@@ -113,21 +115,23 @@ def get_compile_next(self, already_compiled_files: Set[Path], to_compile: Set[An
         return compile_next
 
     def compile_file(self, analysed_file: AnalysedFile):
-        command = [self.exe]
-        command.extend(self.flags.flags_for_path(analysed_file.fpath, self._config.workspace))
-        command.append(str(analysed_file.fpath))
+        with Timer() as timer:
+            command = [self.exe]
+            command.extend(self.flags.flags_for_path(analysed_file.fpath, self._config.workspace))
+            command.append(str(analysed_file.fpath))
 
-        output_fpath = analysed_file.fpath.with_suffix('.o')
-        if self._config.debug_skip and output_fpath.exists():
-            log_or_dot(logger, f'Compiler skipping: {output_fpath}')
-            return CompiledFile(analysed_file, output_fpath)
+            output_fpath = analysed_file.fpath.with_suffix('.o')
+            if self._config.debug_skip and output_fpath.exists():
+                log_or_dot(logger, f'Compiler skipping: {output_fpath}')
+                return CompiledFile(analysed_file, output_fpath)
```

Comment:
Does the timer work correctly with the premature return?

Reply:
Yep, _"The with statement guarantees that if the __enter__() method returns without an error, then __exit__() will always be called."_ [[docs]](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement)